Sources/MapboxNavigation/StyleManager.swift

```diff
@@ -179,6 +179,7 @@ open class StyleManager {
         guard currentStyleType != styleType else { return }
         
         NSObject.cancelPreviousPerformRequests(withTarget: self, selector: #selector(timeOfDayChanged), object: nil)
+        ImageRepository.shared.resetImageCache(nil)
```

Comment:
Not sure that this allows to resolve reported issue. Here's behavior with cache resetting:

https://user-images.githubusercontent.com/1496498/118257487-8c16d280-b463-11eb-8e81-fa78c78afe2c.mov

As you can see icons still remain in color of previously used style. I'd recommend looking into whether cache resetting really removes files, and if so whether removing really correct approach. Maybe there will be need to reload list manually etc.

Reply:
Yes, now it's not working to fix. i'll continue investing this issue to fix.